seqr/views/apis/anvil_workspace_api.py

```diff
@@ -1,57 +1,119 @@
-         Redirect to '/create_project_from_workspace/:workspace_name' if project doesn't exist.
-         Redirect to '/project/:projectGuid'
+    Validate the workspace and project before loading data.
+
+    :param request: Django request object.
+    :param namespace: The namespace (or the billing account) of the workspace.
+    :param name: The name of the workspace. It also be used as the project name.
+    :return Redirect to a page depending on if the workspace permissions or project exists.
 
     """
-    # To be implemented
-    return create_json_response({})
+    check_workspace_perm(request.user, CAN_EDIT, namespace, name, can_share=True)
+
+    project_name = _get_project_name(namespace, name)
+    project = Project.objects.filter(name=project_name)
+    if project:
+        return redirect('/project/{}/project_page'.format(project.first().guid))
+    else:
+        return redirect('/create_project_from_workspace/{}/{}'.format(namespace, name))
 
 
 @login_required(login_url=API_LOGIN_REQUIRED_URL)
 def create_project_from_workspace(request, namespace, name):
     """
-    Create a project when a cooperator requesting to load data from an AnVIL workspace.
+    Create a project when a cooperator requests to load data from an AnVIL workspace.
 
-    Args:
-        request (object): Django request object
-        namespace: The namespace (or the billing account) of the workspace
-        name: The name of the workspace. It also be used as the project name
-    Return:
-        The projectsByGuid with the new project json
+    :param request: Django request object
+    :param namespace: The namespace (or the billing account) of the workspace
+    :param name: The name of the workspace. It also be used as the project name
+    :return the projectsByGuid with the new project json
 
     """
-    response_json = {}  # to be done
-    # Todo:
-    # 1) Validate that the current user has logged in through google and has one of the valid can_edit levels of
-    #  access on the specified workspace;
-    # 2) Validate all the user input from the post body;
-    # 3) Add the seqr service account to the corresponding AnVIL workspace, so that our team will have access to the
-    # project for data loading;
-    # 4) Create a new Project in seqr. This project should NOT be added to the analyst group. The project name should
-    #  just be the workspace name. Make sure to set workspace_namespace and workspace_name correctly;
-    # 5) Add families/individuals based on the uploaded pedigree file;
-    # 6) Send an email to all seqr data managers saying a new AnVIL project is ready for loading. Include the seqr
-    #  project guid, the workspace name, and attach a txt file with a list of the individual IDs that were created.
-    return create_json_response(response_json)
+    project_name = _get_project_name(namespace, name)
+    project = Project.objects.filter(name=project_name)
```

Comment:
It seems that the project created by a collaborator can't be deleted by a Data Manager or a Superuser. If the collaborator created a project for a workspace incorrectly (e.g. having a wrong pedigree data file), how can we help the user solve the problem?

Reply:
So the word "collaborator" in seqr means something very specific, it is a read-only user on a project. They have no ability to create projects at all. I don't think this is who you mean. I assume that in your comment you are talking about "anvil workspace owners who create a project in seqr for which they become a manager", so thats hwo I'm going to answer, but let me know if I misunderstood

First off, superusers have access to the database admin, which means we can edit any data we want, including deleting a whole project (or more realistically editing a project genome that was entered incorrectly). Second, the solution to having incorrect families/individuals in your project is never to delete the whole project, it is to fix the data. Our PMs will have the ability to do this the way they do with all our other projects, via the project page UI

Reply:
Yes, the `collaborator` in my text means an Anvil workspace owner who created a project in seqr. Before the project is created, the user is not required to be a manager. Am I right?

I see what you mean about deleting a project. I did try to delete the project from the database shell but I failed because of the families/individuals data in the project. I know how to do it now. Thank you.